code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -34,94 +34,501 @@
+			if(do_after(user, 25, target = user))
+				empulse(user, 1, 2)
+				user.visible_message("<span class='warning'>[user]'s skin flashes!</span>", "<span class='warning'>Your skin flashes as you emit an EMP pulse!</span>")
+				return 600
+
 /obj/item/slime_extract/red
 	name = "red slime extract"
 	icon_state = "red slime extract"
 
+/obj/item/slime_extract/red/activate(mob/living/carbon/human/user, datum/species/jelly/luminescent/species, activation_type)
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			to_chat(user, "<span class='notice'>You activate [src]. You start feeling fast!</span>")
+			user.reagents.add_reagent("ephedrine",5)
+			return 450
+
+		if(SLIME_ACTIVATE_MAJOR)
+			user.visible_message("<span class='warning'>[user]'s skin flashes red for a moment...</span>", "<span class='warning'>Your skin flashes red as you emit rage-inducing pheromones...</span>")
+			for(var/mob/living/simple_animal/slime/slime in viewers(get_turf(user), null))
+				slime.rabid = 1
+				slime.visible_message("<span class='danger'>The [slime] is driven into a frenzy!</span>")
+			return 600
+
 /obj/item/slime_extract/blue
 	name = "blue slime extract"
 	icon_state = "blue slime extract"
 
+/obj/item/slime_extract/blue/activate(mob/living/carbon/human/user, datum/species/jelly/luminescent/species, activation_type)
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			to_chat(user, "<span class='notice'>You activate [src]. Your genome feels more stable!</span>")
+			user.adjustCloneLoss(-15)
+			user.reagents.add_reagent("mutadone", 10)
+			user.reagents.add_reagent("potass_iodide", 10)
+			return 250
+
+		if(SLIME_ACTIVATE_MAJOR)
+			var/location = get_turf(user)
+			var/datum/effect_system/foam_spread/s = new()
+			s.set_up(20, location, user.reagents)
+			s.start()
+			user.reagents.clear_reagents()
+			user.visible_message("<span class='danger'>Foam spews out from [user]'s skin!</span>", "<span class='warning'>You activate [src], and foam bursts out of your skin!</span>")
+			return 600
+
 /obj/item/slime_extract/darkblue
 	name = "dark blue slime extract"
 	icon_state = "dark blue slime extract"
 
+/obj/item/slime_extract/darkblue/activate(mob/living/carbon/human/user, datum/species/jelly/luminescent/species, activation_type)
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			to_chat(user, "<span class='notice'>You activate [src]. You start feeling colder!</span>")
+			user.ExtinguishMob()
+			user.adjust_fire_stacks(-20)
+			user.reagents.add_reagent("frostoil",5)
+			user.reagents.add_reagent("cryoxadone",5)
+			return 100
+
+		if(SLIME_ACTIVATE_MAJOR)
+			var/turf/open/T = get_turf(user)
+			if(istype(T))
+				T.atmos_spawn_air("nitrogen=40;TEMP=2.7")
+			to_chat(user, "<span class='warning'>You activate [src], and icy air bursts out of your skin!</span>")
```

Comment:
The slimeperson is just going to die without thermal protection, is this intended for self denial or just poorly thought through.

Reply:
If you use the cold-themed core as a slimeperson you get hurt